scripts/run_es_toolkit_validation.py

```
@@ -0,0 +1,398 @@
+) -> tuple[Path, tempfile.TemporaryDirectory[str] | None]:
+  if package_dir is not None and tarball_cache is not None:
+    raise ValueError("--package-dir and --tarball-cache cannot be used together")
+  if package_dir is not None:
+    root = package_dir.resolve()
+    temporary = None
+  else:
+    temporary = tempfile.TemporaryDirectory(prefix="goccia-es-toolkit-")
+    if tarball_cache is not None and tarball_cache.is_file():
+      data = tarball_cache.read_bytes()
+    else:
+      with urllib.request.urlopen(manifest["upstream"]["tarball"], timeout=30) as response:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)